core/bugnote_api.php

```diff
@@ -753,10 +753,38 @@ function bugnote_clear_cache( $p_bugnote_id = null ) {
 
 	if( null === $p_bugnote_id ) {
 		$g_cache_bugnote = array();
+		$g_cache_bugnotes = array();
 	} else {
-		unset( $g_cache_bugnote[(int)$p_bugnote_id] );
+		if( isset( $g_cache_bugnote[(int)$p_bugnote_id] ) ) {
+			$t_note_obj = $g_cache_bugnote[(int)$p_bugnote_id];
+			bugnote_clear_bug_cache( $t_note_obj->bug_id );
+			unset( $g_cache_bugnote[(int)$p_bugnote_id] );
```

Comment:
I think he means that in bugnote_clear_bug_cache(), all entries in $g_cache_bugnote for the corresponding bug_id are already unset (see [line 783](https://github.com/mantisbt/mantisbt/pull/941/files#diff-af4b0457065b17f59cdc20eec8005bc4R783))